zebra/zebra_ptm.c

```diff
@@ -625,28 +625,24 @@ int zebra_ptm_sock_read(struct thread *thread)
 		return -1;
 
 	/* PTM communicates in CSV format */
-	while (!done) {
+	do {
 		rc = ptm_lib_process_msg(ptm_hdl, sock, ptm_cb.in_data,
 					 ZEBRA_PTM_MAX_SOCKBUF, NULL);
-		if (rc <= 0)
-			break;
-	}
+	} while (rc <= 0);
 
-	if (rc <= 0) {
-		if (((rc == 0) && !errno)
-		    || (errno && (errno != EWOULDBLOCK) && (errno != EAGAIN))) {
-			zlog_warn("%s routing socket error: %s(%d) bytes %d",
-				  __func__, safe_strerror(errno), errno, rc);
-
-			close(ptm_cb.ptm_sock);
-			ptm_cb.ptm_sock = -1;
-			zebra_ptm_reset_status(0);
-			ptm_cb.t_timer = NULL;
-			thread_add_timer(zebrad.master, zebra_ptm_connect, NULL,
-					 ptm_cb.reconnect_time,
-					 &ptm_cb.t_timer);
-			return (-1);
-		}
+	if (((rc == 0) && !errno)
```

Comment:
What happens if errno is not cleared from a previous call failure?  Why is this code preferable?  I am not sure what we are trying t odo here

Reply:
Oh, the loop condition is wrong, my fault, sorry.

Reply:
I've fixed the loop condition, now the behavior is exactly like the original.

Reply:
(the only purpose was to remove the redundant code)